api/src/main/java/eu/europeana/fulltext/api/web/FTRetrievalController.java

```diff
@@ -94,6 +96,14 @@ private ResponseEntity<String> getAnnoPageInfo(String datasetId, String localId,
             return cached;
         }
         HttpHeaders headers = CacheUtils.generateHeaders(request, eTag, CacheUtils.zonedDateTimeToString(modified));
+        // by default content-type should be application/ld+json unless accept header is passed as application/json
+        boolean isJson = false;
+        if (request.getHeader(ACCEPT) != null && request.getHeader(ACCEPT).equals(MEDIA_TYPE_JSON)) {
```

Comment:
This is tricky; Accept headers can combine multiple types, look at for example what browsers generally sent; e.g. `text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8` Now, accommodating also priorities is probably a stretch too far, but we could also change this into 'contains()` instead of equals. Perhaps also converting to lowercase first (not sure if Spring Boot does that automatically when sending accept headers with capital letters).

Reply:
I had a look at how we solved this in IIIF Manifest and there we simply created 2 GetMappings, one for json and one for json-ld, so Spring-Boot does the content negotiation work for us

    @GetMapping(value = "/{collectionId}/{recordId}/manifest", headers = ACCEPT_JSON)
    @GetMapping(value = "/{collectionId}/{recordId}/manifest", headers = ACCEPT_JSONLD)
and we set priority in WebConfig which should have preference when we get Accept:*/* but I see that's already present in Fulltext.

Reply:
separated the methods to json and jsonld